src/credentials.ts

```diff
@@ -0,0 +1,76 @@
+import { CredentialDefinition, CredentialOffer, CredentialManifestDisplayMapping } from '@jolocom/protocol-ts'
+import { jsonpath } from './util'
+
+export interface DisplayVal {
+  label?: string,
+    key?: string,
+    value?: string
+}
+
+// TODO actually move into jolocom-lib??
+export class CredentialType {
+  type: string
+  def: CredentialDefinition
+
+  constructor(type: string, def: CredentialDefinition) {
+    this.type = type
+    this.def = def
+  }
+
+  display(claim: any) {
```

Comment:
Should add the `claim` type and proper return type (`CredentialDefinition`) i guess

Reply:
I added types to this, and in the process realized it's better to nest the 'display' property, for clarity that those values would be `DisplayVal` things and not just strings like `cd.name` or `cd.schema`